lib/rules/jsx-no-leaked-render.js

```diff
@@ -160,6 +161,18 @@ module.exports = {
           if (isCoerceValidLeftSide || getIsCoerceValidNestedLogicalExpression(leftSide)) {
             return;
           }
+          const variables = variableUtil.variablesInScope(context);
+          const leftSideVar = variableUtil.getVariable(variables, leftSide.name);
+          if (leftSideVar) {
+            const leftSideValue = leftSideVar.defs
+              && leftSideVar.defs.length
+              && leftSideVar.defs[0].node.init
+              && leftSideVar.defs[0].node.init.value;
+            const leftSideType = typeof leftSideValue;
+            if (leftSideType === 'boolean') {
```

Comment:
```suggestion
            if (typeof leftSideValue === 'boolean') {
```
this actually bypasses engine optimizations sometimes, and the extra variable isn't really adding value.